src/OpenConext/Profile/Entity/AuthenticatedUser.php

```diff
@@ -43,13 +45,47 @@
     /**
      * @param AssertionAdapter $assertionAdapter
      * @param EntityId[] $authenticatingAuthorities
+     *
      * @return AuthenticatedUser
+     * @throws RuntimeException
      */
     public static function createFrom(AssertionAdapter $assertionAdapter, array $authenticatingAuthorities)
     {
+        $attributes = [];
+
+        /** @var Attribute $attribute */
+        foreach ($assertionAdapter->getAttributeSet() as $attribute) {
+            $definition = $attribute->getAttributeDefinition();
+
+            // We need to replace the eduPersonTargetedID attribute value as that is a nested NameID attribute.
+            if ($definition->getName() === 'eduPersonTargetedID') {
+
+                /** @var \DOMNodeList[] $eptiValues */
+                $eptiValues = $attribute->getValue();
+                $eptiDomNodeList = $eptiValues[0];
+
+                if (!$eptiDomNodeList instanceof \DOMNodeList || $eptiDomNodeList->length !== 1) {
+                    throw new RuntimeException(
+                        sprintf(
+                            'EPTI attribute must contain exactly one NameID element as value, received: %s',
+                            print_r($eptiValues, true)
```

Comment:
Did you verify that print_r'ing an array of DOMNodeLists yields useful information? Also: I don't feel like digging up the context for this, but could NameIDs include PII (name and institution) which should not end up in the logs? (Exception messages are likely to be logged.)